tests/unit/array/mixins/test_del.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+
+from docarray import DocumentArray, Document
+
+
+@pytest.fixture()
+def docs():
+    return DocumentArray([Document(id=f"{i}") for i in range(1, 10)])
+
+
+@pytest.mark.parametrize(
+    "to_delete",
+    [
+        0,
+        1,
+        4,
+        -1,
+        list(range(1, 4)),
+        [2, 4, 7, 1, 1],
+        slice(0, 2),
+        slice(2, 4),
+        slice(4, -1),
+        [True, True, False],
+        ...,
+    ],
+)
+def test_del_all(docs, to_delete):
+    doc_to_delete = docs[to_delete]
+    del docs[to_delete]
+    assert doc_to_delete not in docs
+
+
+@pytest.mark.parametrize(
+    ["deleted_ids", "expected_ids"],
+    [
+        (["1", "2", "3", "4"], ["5", "6", "7", "8", "9"]),
+        (["2", "4", "7", "1"], ["3", "5", "6", "8", "9"]),
+    ],
+)
```

Comment:
- use `black -s`
- or a better way is to install pre-commit hook via 
```python
# at docarray root
pre-commit install
```

you will see this option auto-checked on every commit
![image](https://user-images.githubusercontent.com/2041322/151313085-dd2600fd-0f2c-49f3-90e7-2da75d2d28bd.png)

Reply:
I already installed the pre-commit hook, this is already formatted by black